src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2312,14 +2484,16 @@ TablePartitioningSupportsDistinct(List *tableNodeList, MultiExtendedOp *opNode,
 		 */
 		partitionMethod = PartitionMethod(relationId);
 
+
 		if (partitionMethod == DISTRIBUTE_BY_RANGE ||
 			partitionMethod == DISTRIBUTE_BY_HASH)
 		{
 			Var *tablePartitionColumn = tableNode->partitionColumn;
 			bool groupedByPartitionColumn = false;
 
 			/* if distinct is on table partition column, we can push it down */
-			if (tablePartitionColumn->varno == distinctColumn->varno &&
+			if (distinctColumn != NULL &&
```

Comment:
We return when it is a subquery above. So, do we actually need this? (I see that this cannot have any side effects, but still, why we needed this?)